### PR TITLE
tar2rpm allow specifying a build time stamp via the cli

### LINKED ISSUE #43: tar2rpm: Build Date is far in the future
The rpm created by tar2rpm contains a "Build Date" that is far in the future.
Steps to reproduce:

kali@kali:~/go$ date
`Wed 08 Jul 2020 04:02:49 PM EDT`

kali@kali:~/go$ ./tar2rpm -file /tmp/example.rpm -name "Example" -version "1.0" /tmp/input.tar

kali@kali:~/go$ rpm -qip /tmp/example.rpm
```
Name        : Example
Epoch       : 0
Version     : 1.0
Release     : 
Architecture: noarch
Install Date: (not installed)
Group       : 
Size        : 727851
License     : 
Signature   : (none)
Source RPM  : Example-1.0.src.rpm
Build Date  : Sun 13 Jul 2042 09:04:00 PM EDT
Build Host  : 
Relocations : (not relocatable)
Packager    : 
Vendor      : 
URL         : 
Summary     : 
Description :
```

Expectation:
The "Build Date:" line of the rpm query should contain something like "Wed 08 Jul 2020 04:02:49 PM EDT", but it's always "Sun 13 Jul 2042 09:04:00 PM EDT" even on subsequent delayed rebuilds of the rpm.

### PR DESCRIPTION
**Pull request by jarondl**
Add a build_time flag which allows specifying the build timestamp for
people who like that.
To use the current time, you can use:
` --build_time $(date +%s)`

I strongly recommend not doing that, because it makes the build change
every time and hurt your caching and reproducibility, but people seem to
need this.

Also change the default timestamp to be empty to avoid the 2042 time.

Fixes #43

### CODE REVIEW

File: cmd/tar2rpm/main.go [hunk 1]
```diff
@@ -83,6 +85,10 @@ func main() {
 		flag.Usage()
 		os.Exit(2)
 	}
+	var buildTimeStamp time.Time
+	if *buildTime != 0 {
```
Comment (alexaguilera93) [inline cmd/tar2rpm/main.go:89]: Is this check really necessary, or can it just be condensed to:

```
buildTimeStamp := time.Unix(*buildTime, 0)
```

File: def.bzl [hunk 2]
```diff
@@ -34,6 +36,7 @@ pkg_tar2rpm = rule(
         "preun": attr.string(),
         "postun": attr.string(),
         "requires": attr.string_list(),
+        "build_time": attr.string(),
```
Comment (alexaguilera93) [inline def.bzl:39]: Maybe just add this with a default value of 0, so there's no need to check for build_time with args.add

Comment (alexaguilera93) [reply def.bzl:39]: Also, it may be worth adding a doc string with the expected format of the build_time.

File: cmd/tar2rpm/main.go (hunk 1 above)
Comment (jarondl) [reply cmd/tar2rpm/main.go:89]: No, surprisingly (as explained in the bug), the zero value of `time.Time` is not the unix epoch. It is the AD 1 year https://en.wikipedia.org/wiki/AD_1 and this ends up as 2042. I'd rather not introduce another zero value here.

File: def.bzl (hunk 2 above)
Comment (jarondl) [reply def.bzl:39]: That's a good idea, but once again, because the zero value is not exactly the zero time, I'd rather not pass the value at all.

**Pull request merged**